src/lib/process.ts

```diff
@@ -0,0 +1,118 @@
+import { EventListener } from "./events";
+
+const RunService = game.GetService("RunService");
+
+export default function processHandler() {
+	const TPS = 20; // TPS to be replaced with value from tina.yaml
+	const processes = new Map<string, Process>();
+	const scheduler = new ProcessScheduler(TPS);
+
+	return (name: string) => {
+		if (processes.has(name)) {
+			return processes.get(name)!;
+		}
+		return new Process(name, scheduler);
+	};
+}
+
+/**
+ *
+ */
+export class Process extends EventListener<[]> {
+	public isSuspended = false;
+	public suspensionTime = -1;
+	public name: string;
+
+	constructor(name: string, private ticker: ProcessScheduler) {
```

Comment:
You're already doing
```ts
constructor(name: string, private ticker: ProcessScheduler) {
```
So why not do
```ts
constructor(public name: string, private ticker: ProcessScheduler) {
```
Let me answer that, stylistically just don't do inline constructor field definitions at all, put the ticker definition in the root.